File: app/src/main/java/net/pmellaaho/rxapp/network/GitHubApi.java
package net.pmellaaho.rxapp.network;

import net.pmellaaho.rxapp.model.Contributor;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface GitHubApi {

    /**
     * See https://developer.github.com/v3/repos/#list-contributors
     */
    @GET("/repos/{owner}/{repo}/contributors")
    Observable<List<Contributor>> contributors(@Path("owner") String owner,
                                               @Path("repo") String repo);

}

File: app/src/main/java/net/pmellaaho/rxapp/network/GitHubApi.kt
package net.pmellaaho.rxapp.network

import net.pmellaaho.rxapp.model.Contributor
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Path

private const val OWNER = "square"

interface GitHubApi {

    /**
     * See https://developer.github.com/v3/repos/#list-contributors
     */
    @Headers("Content-Type: application/json")
    @GET("/repos/{owner}/{repo}/contributors")
    suspend fun getContributors(
        @Path("repo") repo: String,
        @Path("owner") owner: String = OWNER
    ): List<Contributor>
}